sotodlib/preprocess/preprocess_util.py

```diff
@@ -0,0 +1,506 @@
+import os
+import logging
+import time
+import sys
+import copy
+import yaml
+import numpy as np
+
+from .. import core
+
+from . import _Preprocess, Pipeline, processes
+
+class ArchivePolicy:
+    """Storage policy assistance.  Helps to determine the HDF5
+    filename and dataset name for a result.
+
+    Make me better!
+
+    """
+    @staticmethod
+    def from_params(params):
+        if params['type'] == 'simple':
+            return ArchivePolicy(**params)
+        if params['type'] == 'directory':
+            return DirectoryArchivePolicy(**params)
+        raise ValueError('No handler for "type"="%s"' % params['type'])
+
+    def __init__(self, **kwargs):
+        self.filename = kwargs['filename']
+
+    def get_dest(self, product_id):
+        """Returns (hdf_filename, dataset_addr).
+
+        """
+        return self.filename, product_id
+
+
+class DirectoryArchivePolicy:
+    """Storage policy for stuff organized directly on the filesystem.
+
+    """
+    def __init__(self, **kwargs):
+        self.root_dir = kwargs['root_dir']
+        self.pattern = kwargs['pattern']
+
+    def get_dest(self, **kw):
+        """Returns full path to destination directory.
+
+        """
+        return os.path.join(self.root_dir, self.pattern.format(**kw))
```

Comment:
I think I'm a bit ambivalent about this as I think there were some hopes for these classes but they didn't really amount to much and just include the simplest return of h5group and file paths (so its probably fine to leave these here as is). But these are generally used by all manifestdb's not just preprocess so it seems like they could instead go in core.util or core.metadata.util (or were appropriately placed in site_pipeline.util since that's where the manifestdB generation scripts live). Do you care about this at all @mhasself or @kmharrington ?